test/store_unittest.cc

```diff
@@ -0,0 +1,226 @@
+      json::object({{"f", json::string("hello")}}));
+  MetadataStore::Metadata m2(
+      "default-version2",
+      false,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  store.UpdateMetadata(mr1, std::move(m1));
+  store.UpdateMetadata(mr2, std::move(m2));
+  const auto metadata_map = GetMetadataMap();
+  EXPECT_FALSE(metadata_map.empty());
+  EXPECT_EQ("default-version1", metadata_map.at(mr1).version);
+  EXPECT_EQ("default-version2", metadata_map.at(mr2).version);
+}
+
+TEST_F(MetadataStoreTest, UpdateMetadataEntry) {
+  MonitoredResource mr("some_resource", {});
+  MetadataStore::Metadata m1(
+      "default-version1",
+      false,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  store.UpdateMetadata(mr, std::move(m1));
+  const auto metadata_map_before = GetMetadataMap();
+  EXPECT_FALSE(metadata_map_before.empty());
+  EXPECT_EQ("default-version1", metadata_map_before.at(mr).version);
+  MetadataStore::Metadata m2(
+      "default-version2",
+      false,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  store.UpdateMetadata(mr, std::move(m2));
+  const auto metadata_map_after = GetMetadataMap();
+  EXPECT_EQ("default-version2", metadata_map_after.at(mr).version);
+}
+
+TEST_F(MetadataStoreTest, PurgeDeletedEntries) {
+  MonitoredResource mr1("some_resource1", {});
+  MonitoredResource mr2("some_resource2", {});
+  MetadataStore::Metadata m1(
+      "default-version1",
+      false,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  MetadataStore::Metadata m2(
+      "default-version2",
+      true,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  store.UpdateMetadata(mr1, std::move(m1));
+  store.UpdateMetadata(mr2, std::move(m2));
+  const auto metadata_map_before = GetMetadataMap();
+  EXPECT_FALSE(metadata_map_before.empty());
+  EXPECT_EQ("default-version1", metadata_map_before.at(mr1).version);
+  EXPECT_EQ("default-version2", metadata_map_before.at(mr2).version);
+  PurgeDeletedEntries();
+  const auto metadata_map_after = GetMetadataMap();
+  EXPECT_EQ("default-version1", metadata_map_after.at(mr1).version);
+  EXPECT_THROW(metadata_map_after.at(mr2), std::out_of_range);
+}
+
+TEST(MetadataTest, ParseMetadata) {
+  MetadataStore::Metadata m(
+      "default-version",
+      false,
+      time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+      time::rfc3339::FromString("2018-03-03T01:32:45.678901234Z"),
+      json::object({{"f", json::string("hello")}}));
+  EXPECT_EQ("default-version", m.version);
+  EXPECT_FALSE(m.is_deleted);
+  EXPECT_FALSE(m.ignore);
```

Comment:
Let's make sure to test this first, and then test the contents.

Reply:
Done.